lib/ex_machina/ecto_strategy.ex

```diff
@@ -74,16 +74,17 @@ defmodule ExMachina.EctoStrategy do
     assocs = get_schema_assocs(schema)
 
     Enum.reduce(assocs, struct, fn(assoc, struct) ->
-      original_value = Map.get(struct, assoc)
-      casted_value = if is_list(original_value) do
-        Enum.map(original_value, &(cast_assoc(&1, assoc, struct)))
-      else
-        cast_assoc(original_value, assoc, struct)
-      end
+      casted_value = Map.get(struct, assoc)
+                   |> cast_assoc(assoc, struct)
+
       Map.put(struct, assoc, casted_value)
     end)
   end
 
+  defp cast_assoc(original_value, assoc, struct) when is_list(original_value) do
```

Comment:
Yeah, I was thinking about that too. I guess the difference is that here the `original_value` is a list as opposed to an map. The rest of the branching logic is dealing maps, one way or another, so it fits more nicely. That being said, I can see why we'd want it in a single place. 

Do you think this is a preferable? 

```elixir
case original_value do 
  original_value when is_list(originaLvalue) -> 
    Enum.map(original_value, &(cas_assoc(&1, assoc, struct)))
  
  %{__meta__: %{__struct__: Ecto.Schema.Metadata, state: :built}} ->
    cast(original_value)

  %{__struct__: Ecto.Association.NotLoaded} ->
    original_value

  %{__struct__: _} ->
    cast(original_value)

  %{} ->
    
# ...

```